Removed some of the UTC logic which shouldn't be there

emrysal opened the pull request:
## What does this PR do?

Fixes 11:00 - 11:00 All Day date override display issue.

* Upgrades RHF - this version supports async values so we can start getting rid of useEffect @zomars 
* Pulls in some changes from another PR
* Remove getServerSideProps
* Sorts date overrides by ascending date
* Fixed bug that caused date overrides that were still in the future to be hidden from the listing.

emrysal committed:
Removed some of the UTC logic which shouldn't be there

emrysal commented on packages/features/schedules/components/DateOverrideList.tsx:
```diff
@@ -50,7 +48,7 @@ const DateOverrideList = ({
               }).format(item.ranges[0].start)}
             </h3>
-            {item.ranges[0].end.getUTCHours() === 0 && item.ranges[0].end.getUTCMinutes() === 0 ? (
+            {item.ranges[0].end.getHours() === 0 && item.ranges[0].end.getMinutes() === 0 ? (
```
getHours/getMinutes being 0-0 is only possible during an all-day event.

CarinaWolli commented:
It shows 7:00 PM - 7:00 PM now for me 🤔
![Screenshot 2023-02-27 at 11 13 14](https://user-images.githubusercontent.com/30310907/221618196-7dac808a-f9e4-483b-879c-866efe7ac893.png)

emrysal committed:
Upgrade rhf, remove SSR on [schedule] page

emrysal committed:
Fix potential bug with date

emrysal committed:
Amend schedule trpc call & don't show edit availability if id is undefined

emrysal committed:
Merge branch 'main' into bugfix/11-00-11-00-dst-issue

emrysal committed:
type fix

emrysal committed:
Order date overrides in list by date

emrysal committed:
Changed logic to check for unavailable

CarinaWolli commented on packages/trpc/server/routers/viewer/availability.tsx:
```diff
@@ -157,15 +92,51 @@ export const availabilityRouter = router({
           timeZone: schedule.timeZone || user.timeZone,
+          dateOverrides: schedule.availability.reduce((acc, override) => {
+            // only iff future date override
+            if (!override.date || override.date < new Date()) {
```
I can save date overrides for march 1st and 2nd (my date/time saving was march 1st at 11PM GMT-5). It is saved successfully in the database however the date overrides don't show up in the list. 

That's why:
- `new Date()` is utc so it is already march 2nd
- `override.date` has always timestamp 00:00 so it is also smaller than the utc time which was in my case 2023-03-02T04:00

emrysal replied:
Nice find this bug! Fixed it now ⛅

zomars reviewed (approved):
Async values looks like a nice improvement. Thanks @emrysal 🙏🏽

emrysal committed:
Fix bug report spotted by @CarinaWolli

I can save date overrides for march 1st and 2nd (my date/time when saving was march 1st at 11PM GMT-5). It is saved successfully in the database however the date overrides don't show up in the list.

That's why:

- new Date() is utc so it is already march 2nd
- override.date has always timestamp 00:00 so it is also smaller than the utc time which was in my case 2023-03-02T04:00

emrysal committed:
Merge branch 'main' into bugfix/11-00-11-00-dst-issue

emrysal committed:
Fixed merge conflicts, removed unused function

emrysal committed:
Actual fix, gotta remember it's the schedule tz
